### PR TITLE
Reader: make it so that you cannot search for just whitespace

### PR DESCRIPTION
**Pull request by samouri**
Currently the search page in the Reader allows you to search for whitespace alone which causes a weird UX. More details on the issue can be found at https://github.com/Automattic/wp-calypso/issues/10794.

It turns out the condition to fix this was already in the if statement but missing an exclamation point :D

### CODE REVIEW

**Comment by bluefuton** [comment]
Looks like this breaks clearing of the search box. If you input a search string and then click the 'x' , you're not returned to the default state (recommendations).

**Comment by samouri** [comment]
good catch @bluefuton! I'll fix that

**Comment by samouri** [comment]
To Test:
- search around in reader search.  make sure to click suggestions as well as search empty strings
- make sure other searches aren't broken (specifically w.r.t initial value) because I needed to fix a bug in there

**Comment by samouri** [comment]
currently getting this error:
`You provided a `value` prop to a form field without an onChange`

Will need to investigate more

**Comment by samouri** [comment]
From my investigation:
- React wants us to use `onChange` instead of `onInput`.  They are the [same thing](https://github.com/facebook/react/pull/4003/files).
- Thats all.

File: client/reader/search-stream/index.jsx [hunk 1]
```diff
@@ -194,16 +199,16 @@ class SearchStream extends Component {
...
+		const emptyContent = <EmptyContent query={ query } />;
 
 		let searchPlaceholderText = this.props.searchPlaceholderText;
 		if ( ! searchPlaceholderText ) {
 			searchPlaceholderText = this.props.translate( 'Search billions of WordPress.com posts…' );
 		}
 
-		const sugList = initial( flatMap( this.state.suggestions, query =>
-			[ <Suggestion suggestion={ query } source="search" />, ', ' ] ) );
+		const sugList = initial( flatMap( this.state.suggestions, q =>
```
Comment (bluefuton) [inline client/reader/search-stream/index.jsx]: Is there a more descriptive variable name we could use instead of `q`?

Comment (samouri) [inline client/reader/search-stream/index.jsx]: sure.  I'll rename to `suggestionKeyword`

**Comment by samouri** [comment]
While testing I noticed its currently broken for the search on following page ab test.  
Fixing

**Pull request merged**